ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -165,6 +165,26 @@ class JITModBuilder {
     }
   }
 
+  Function *createJITFunc(vector<Value *> *TraceInputs) {
+    // Compute a name for the trace.
+    uint64_t TraceIdx = NextTraceIdx.fetch_add(1);
+    if (TraceIdx == numeric_limits<uint64_t>::max())
```

Comment:
Same comment as before (this is probably better as a debug `assert` before the `fetch_add`).

Reply:
Shouldn't we do this check always? An assert would only do the check in debug builds.

Reply:
It takes something like 100 years of continuous incrementing for a 64 bit int to overflow on a single core machine, so having a permanent assert for this seems a bit OTT: it's, ahem, not very likely to trigger. That said, I *like* debug asserts because they can catch other weirder errors (e.g. someone accidentally sets this to u64::max() or similar), but without a runtime cost.

Reply:
So on bencher13, we could overflow that integer in just under 3-years (unreasonable assuming that a) there's 0-cost threading overhead, and b) a program could spawn so much JIT compilation. But you get my point).

When this integer overflows the program is likely to explode anyway (probably with a  symbol linkage error), so the question then becomes: when do you want it to explode? When the invalid ID is forged, or later?

Reply:
There is not zero threading overhead and it couldn't spawn that much JIT compilation ;) If you managed to spill this in our case in under 100,000 years, I'd be flabbergasted. It's not going to explode in anyone's lifetimes *unless* someone does something idiotic, at which point a debug assert should help catch that.

Reply:
> at which point a debug assert should help catch that

Why would a user running a JIT for years be using a debug build though?

I have a better idea (for a later PR).

The trace's name is only useful for debugging (for setting breakpoints in gdb). So we can only assign a name to the trace in debug builds. In release builds we can let LLVM do whatever it does to de-duplicate JIT symbol names. Sounds good?

(end extreme bikeshedding)

Reply:
> Why would a user running a JIT for years be using a debug build though?

We're literally talking tens of thousands years, minimum, for this to happen. No-one will have a single yk process running for that long ;)

> The trace's name is only useful for debugging (for setting breakpoints in gdb). So we can only assign a name to the trace in debug builds. In release builds we can let LLVM do whatever it does to de-duplicate JIT symbol names. Sounds good?

Works for me!

Reply:
https://github.com/ykjit/yk/issues/402

Reply:
OK. Let's at least put that stuff in a function is this PR, though, so addressing the issue later becomes easier?

Reply:
Other stuff fixed in 15e019e7c7679